File: Labtwo/RUBY/pr3.rb
def funcnum(numb)
  # Проверка что все символы - цифры
  return unless numb =~ /^\d+$/
  
  # Проверка длины
  return if numb.length < 2
  
  num = numb.to_i
  sum = 0
  proizv = 1
  
  print "#{num}: "
  
  # Сумма цифр
  numb.chars.each_with_index do |c, i|
    d = c.to_i
    sum += d
    print c
    print "+" if i < numb.length - 1
  end
  print "=#{sum}, "
  
  # Произведение цифр
  numb.chars.each_with_index do |c, i|
    d = c.to_i
    proizv *= d
    print c
    print "*" if i < numb.length - 1
  end
  print "=#{proizv}, "
  
  # Модуль разности
  diff = (sum - proizv).abs
  puts "|#{sum}-#{proizv}|=#{diff}"
end

puts "Write your number"
input = gets.chomp
input.split.each { |numb| funcnum(numb) }

File: Labtwo/RUST/pr3.rs
use std::io;

// Функция обработки числовой строки
fn funcnum(numb: &str) {
    // Проверка что все символы - цифры
    if !numb.chars().all(|c| c.is_ascii_digit()) {
        return;
    }
    
    // Проверка длины числа
    if numb.len() < 2 {
        return;
    }
    
    // Парсинг числа
    let num: i32 = numb.parse().unwrap();
    let mut sum = 0;
    let mut proizv = 1;
    
    // Вычисление суммы и произведения
    let digits: Vec<char> = numb.chars().collect();
    print!("{}: ", num);
    
    // Сумма цифр
    for (i, c) in digits.iter().enumerate() {
        let d = c.to_digit(10).unwrap() as i32;
        sum += d;
        print!("{}", d);
        if i < digits.len() - 1 {
            print!("+");
        }
    }
    print!("={}, ", sum);
    
    // Произведение цифр
    for (i, c) in digits.iter().enumerate() {
        let d = c.to_digit(10).unwrap() as i32;
        proizv *= d;
        print!("{}", d);
        if i < digits.len() - 1 {
            print!("*");
        }
    }
    print!("={}, ", proizv);
    
    // Модуль разности
    let diff = (sum - proizv).abs();
    println!("|{}-{}|={}", sum, proizv, diff);
}

fn main() {
    println!("Write your number");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    
    // Обработка каждого числа в строке
    for numb in input.split_whitespace() {
        funcnum(numb);
    }
}